diff --git a/node_modules/react-native-image-marker/android/src/main/java/com/jimmydaddy/imagemarker/base/MarkTextOptions.kt b/node_modules/react-native-image-marker/android/src/main/java/com/jimmydaddy/imagemarker/base/MarkTextOptions.kt
index 31b3043..7393c47 100644
--- a/node_modules/react-native-image-marker/android/src/main/java/com/jimmydaddy/imagemarker/base/MarkTextOptions.kt
+++ b/node_modules/react-native-image-marker/android/src/main/java/com/jimmydaddy/imagemarker/base/MarkTextOptions.kt
@@ -12,7 +12,7 @@ class MarkTextOptions(options: ReadableMap) : Options(options) {
       watermarkTexts = arrayOfNulls(waterMarkTextsMap.size())
       for (i in 0 until waterMarkTextsMap.size()) {
         val textMap = waterMarkTextsMap.getMap(i)
-        watermarkTexts[i] = TextOptions(textMap)
+        watermarkTexts[i] = textMap?.let { TextOptions(it) }
       }
     }
   }
